### PR TITLE
[Fix] Do not depend on setuptools>=52

### PR DESCRIPTION
**Pull request by kennymckormick**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import glob
 import os
 import re
-from pkg_resources import DistributionNotFound, get_distribution, parse_version
+from pkg_resources import DistributionNotFound, get_distribution
 from setuptools import find_packages, setup
 
+from packaging.version import parse
```
Comment (zhouzaida) [inline setup.py:7]:
```suggestion
from packaging.version import parse as parse_version
```

**Comment by innerlee** [comment]
See concerns here https://github.com/open-mmlab/mmcv/pull/1185#discussion_r669379489

**Comment by kennymckormick** [comment]
@zhouzaida  CI passed

File: mmcv/version.py [hunk 2]
```diff
@@ -32,6 +30,6 @@ def parse_version_info(version_str: str, length: int = 4) -> tuple:
     return tuple(release)
 
 
-version_info = parse_version_info(__version__)
+version_info = tuple([int(x) for x in __version__.split('.')])
```
Comment (zhouzaida) [inline mmcv/version.py:33]:
```suggestion
version_info = tuple(int(x) for x in __version__.split('.'))
```
In addition, if `__version__` contains non-numeric string like `1.3.10.dev`, `int(x)` will throw an error

**Review by ZwwWayne — APPROVED** [review]

File: mmcv/version.py [hunk 3]
```diff
@@ -32,6 +30,6 @@ def parse_version_info(version_str: str, length: int = 4) -> tuple:
     return tuple(release)
 
 
-version_info = parse_version_info(__version__)
+version_info = tuple(int(x) for x in __version__.split('.')[:3])
```
Comment (ZwwWayne) [inline mmcv/version.py:33]: Why do not use parse_version_info here though you wrote it?

Comment (kennymckormick) [reply mmcv/version.py:33]: When you execute this line during setup, "packaging" may not exist.

**Pull request merged**